src/edu/washington/escience/myria/util/SQLiteUtils.java

```diff
@@ -119,4 +119,81 @@ public static void createTable(final String dbFileAbsolutePath, final RelationKe
       }
     }
   }
+
+  /**
+   * Deletes a relation in SQLite.
+   * 
+   * @throws SQLiteException if SQLite error occur
+   * @param dbFileAbsolutePath the SQLite file absolute path
+   * @param relationKey the relation key to delete
+   * */
+  public static void deleteTable(final String dbFileAbsolutePath, final RelationKey relationKey) throws IOException,
+      SQLiteException {
+    SQLiteConnection sqliteConnection = null;
+    SQLiteStatement statement = null;
+    try {
+      final File f = new File(dbFileAbsolutePath);
+
+      if (!f.getParentFile().exists()) {
+        f.getParentFile().mkdirs();
+      }
+
+      /* Connect to the database */
+      sqliteConnection = new SQLiteConnection(f);
+      sqliteConnection.open(true);
+
+      statement =
+          sqliteConnection
+              .prepare("drop table if exists " + relationKey.toString(MyriaConstants.STORAGE_SYSTEM_SQLITE));
+      statement.step();
+
+    } finally {
+      if (statement != null) {
+        statement.dispose();
+      }
+      if (sqliteConnection != null) {
+        sqliteConnection.dispose();
+      }
+    }
+  }
+
+  /**
+   * Checks if a relation exists in the SQLite database
+   * 
+   * @throws SQLiteException if SQLite error occur
+   * @param dbFileAbsolutePath the SQLite file absolute path
+   * @param relationKey the relation key to check
+   * */
+  public static boolean existsTable(final String dbFileAbsolutePath, final RelationKey relationKey) throws IOException,
+      SQLiteException {
+    boolean exists = false;
+    SQLiteConnection sqliteConnection = null;
+    SQLiteStatement statement = null;
+    try {
+      final File f = new File(dbFileAbsolutePath);
+
+      if (!f.getParentFile().exists()) {
```

Comment:
How could this ever be a valid failure scenario? Shouldn't any caller be able to assume that the database file exists?